sling.go

```diff
@@ -462,18 +453,15 @@ func (s *Sling) Do(ctx context.Context) (*http.Response, error) {
 	return resp, nil
 }
 
-func (s *Sling) doWithTrace(req *http.Request) (*http.Response, error) {
-	if s.tracer != nil {
-		err := s.tracer.BeginTrace(req.Context())
-		if err != nil {
-			return nil, err
-		}
-	}
-
+func (s *Sling) do(req *http.Request) (*http.Response, error) {
 	resp, err := s.httpClient.Do(req)
-
-	if resp != nil && s.tracer != nil {
-		resp.Body = &bodyWithTracer{req.Context(), resp.Body, s.tracer, false}
+	if err != nil {
+		if err == context.Canceled {
+			ctxErr := context.Cause(req.Context())
+			if ctxErr != nil {
+				err = ctxErr
+			}
+		}
 	}
```

Comment:
Go 1.20 has added `context.Cause` that lets context errors be more specific. This enables it to work in Sling.

Reply:
Nice